apps/auth/src/users/users.service.ts

```diff
@@ -33,7 +32,17 @@ export class UsersService {
         return this.usersRepository.create(user);
     }
 
-    async verifyUser(email: string, password: string) {
+    private async validateCreateUserDto(createUserDto: CreateUserDto) {
+        try {
+            await this.usersRepository.findOne({ email: createUserDto.email});
+        } catch (err) {
+            return;
+        }
+
+        throw new UnprocessableEntityException('Email already exists');
```

Comment:
is it ok that we throw UnprocessableEntityException everytime when we call validateCreateUserDto?

Reply:
let's move it to try block